datahub-web-react/src/app/entity/shared/tabs/Dataset/Validations/utils.tsx

```diff
@@ -0,0 +1,324 @@
+        enabled: true,
+        visible: true,
+        requiresConnectionSupportedByMonitors: false,
+    },
+    {
+        name: 'Schema',
+        description: "Define & monitor your expectations about the table's columns and their types",
+        icon: <StyledCodeOutlined />,
+        type: AssertionType.DataSchema,
+        entityTypes: [EntityType.Dataset],
+        enabled: true,
+        visible: true,
+    },
+    {
+        name: 'Custom',
+        description: 'Define & monitor your expectations using custom SQL rules',
+        icon: <StyledConsoleSqlOutlined />,
+        type: AssertionType.Sql,
+        entityTypes: [EntityType.Dataset],
+        enabled: true,
+        visible: true,
+        requiresConnectionSupportedByMonitors: true,
+    },
+    {
+        name: 'Other',
+        description: 'Assertions that are defined and maintained outside of DataHub.',
+        icon: <StyledApiOutlined />,
+        type: AssertionType.Dataset,
+        entityTypes: [EntityType.Dataset],
+        enabled: false,
+        visible: false,
+    },
+];
+
+const ASSERTION_TYPE_TO_INFO = new Map();
+ASSERTION_INFO.forEach((info) => {
+    ASSERTION_TYPE_TO_INFO.set(info.type, info);
+});
+
+const getAssertionGroupName = (type: AssertionType): string => {
+    return ASSERTION_TYPE_TO_INFO.has(type) ? ASSERTION_TYPE_TO_INFO.get(type).name : 'Unknown';
+};
+
+const getAssertionGroupTypeIcon = (type: AssertionType) => {
+    return ASSERTION_TYPE_TO_INFO.has(type) ? ASSERTION_TYPE_TO_INFO.get(type).icon : undefined;
+};
+
+/**
+ * Returns a status summary for the assertions associated with a Dataset.
+ *
+ * @param assertions The assertions to extract the summary for
+ */
+export const getAssertionsSummary = (assertions: Assertion[]): AssertionStatusSummary => {
+    const summary = {
+        passing: 0,
+        failing: 0,
+        erroring: 0,
+        total: 0,
+        totalAssertions: assertions.length,
+    };
+    assertions.forEach((assertion) => {
+        if ((assertion.runEvents?.runEvents?.length || 0) > 0) {
+            const mostRecentRun = assertion.runEvents?.runEvents?.[0];
+            const resultType = mostRecentRun?.result?.type;
+            if (AssertionResultType.Success === resultType) {
+                summary.passing++;
+            }
+            if (AssertionResultType.Failure === resultType) {
+                summary.failing++;
+            }
+            if (AssertionResultType.Error === resultType) {
+                summary.erroring++;
+            }
+            if (AssertionResultType.Init !== resultType) {
+                summary.total++; // only count assertions for which there is one completed run event, ignoring INIT statuses!
+            }
+        }
+    });
+    return summary;
+};
+
+/**
+ * TODO: We will remove this mapping code once we replace the OSS legacy assertions summary with the new
+ * format.
+ */
+export const getLegacyAssertionsSummary = (assertions: Assertion[]) => {
```

Comment:
is this still used? can we drop this? 
if it is used we should update the comment so it's clean. We don't want to expose SaaS comments with public OSS code
`TODO: remove this mapping code once legacy assertions summary is updated with a newer format`

Reply:
Done